### PR TITLE
auth: move registration behind option

### PR DESCRIPTION
**Pull request by dcramer**
Refs GH-5328

### CODE REVIEW

**Comment by dcramer** [comment]
I will add the UI here as well, and we'll need to coordinate a change on sentry.io to bind this option

**Comment by dcramer** [comment]
This is ready to go

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,6 +6,10 @@
         "static": true,
         "node_modules": true,
         "htmlcov": true,
+        "build": true,
```
Comment (mitsuhiko) [inline .vscode/settings.json]: not sure if you wanted that to be here.

File: src/sentry/utils/auth.py [hunk 2]
```diff
@@ -269,6 +269,12 @@ def log_auth_failure(request, username=None):
     })
 
 
+def has_user_registration():
+    from sentry import features, options
+
+    return features.has('auth:register') and options.get('auth.allow-registration')
```
Comment (mitsuhiko) [inline src/sentry/utils/auth.py]: does this not mean that people that upgrade will have user registration disabled?

Comment (dcramer) [inline src/sentry/utils/auth.py]: @mitsuhiko yes but its exposed in the setup wizard where they can enable it

File: .vscode/settings.json (hunk 1 above)
Comment (dcramer) [inline .vscode/settings.json]: not specifically but its fineeeee

**Review by mitsuhiko — APPROVED** [review]

**Comment by dcramer** [comment]
```
commit a9f918c540b2f8018c62f74f962e90cbcc8d5156
Author: David Cramer <dcramer@gmail.com>
Date:   Mon Jun 26 11:45:31 2017 -0700

    auth: allow registration per new option

diff --git a/getsentry/conf/settings/defaults.py b/getsentry/conf/settings/defaults.py
index 6a7760b..2e52899 100644
--- a/getsentry/conf/settings/defaults.py
+++ b/getsentry/conf/settings/defaults.py
@@ -288,6 +288,7 @@ SENTRY_ALLOW_PROJECT_CREATION = True
 SENTRY_USE_TRENDING = False
 SENTRY_USE_JS_CLIENT = False
 SENTRY_FEATURES['auth:register'] = False
+SENTRY_OPTIONS['auth.allow-registration'] = True
 SENTRY_FEATURES['projects:dsym'] = True
 SENTRY_FEATURES['billing:metered-plans'] = False
```

**Comment by imsickofmaps** [comment]
@dcramer when will the docker image next be re-built to release this change there? Thanks!

**Comment by dcramer** [comment]
@imsickofmaps we tag new versions of sentry at the beginning of every month

**Pull request merged**